include/pybind11/detail/class.h

```diff
@@ -614,10 +614,11 @@ extern "C" inline int pybind11_getbuffer(PyObject *obj, Py_buffer *view, int fla
     if ((flags & PyBUF_FORMAT) == PyBUF_FORMAT) {
         view->format = const_cast<char *>(info->format.c_str());
     }
+    if ((flags & PyBUF_ND) == PyBUF_ND) {
+        view->shape = info->shape.data();
```

Comment:
I think moving `view->ndim = (int) info->ndim;` to this block and restoring the default of `view->ndim = 1;` in the above block would make this work with all implementations, I agree that this disagrees with what is stated in the documentation linked to.